src/cargo/core/source/source_id.rs

```diff
@@ -257,7 +258,11 @@ impl SourceId {
     }
 
     /// Creates an implementation of `Source` corresponding to this ID.
-    pub fn load<'a>(self, config: &'a Config) -> CargoResult<Box<dyn super::Source + 'a>> {
+    pub fn load<'a>(
+        self,
+        config: &'a Config,
+        yanked_whitelist: HashSet<PackageId>,
```

Comment:
It might help clean things up to plumb through `&HashSet<PackageId>` mostly and then only clone when necessary at the bottom

Reply:
I have been experimenting but plumbing thure the lifetime can be annoying. I will try again.

Reply:
Oh and to be clear I wouldn't connect the lifetime to anything, only passing around a borrow and then when it needs to be stored cloning it